Fix ERB trim mode to match on %> instead of <%

obrie opened the pull request:
This is a follow-up to https://github.com/gjtorikian/graphql-docs/pull/41.  Since the `<%` ERB tags are often indented, they don't get matched by ERB's `<>` trim mode option.  In http://docs.rubydocs.org/ruby-2-4-1/classes/ERB.html#method-c-new, the docs declare that the line must *start* with `<%`.

With this fix, we are matching lines that *end* with `%>`.  This will allow ERB to trim those lines properly.

Definitely double check me on this since I appear to have missed this in the last PR :)

@gjtorikian

gjtorikian commented:
I added a test for this (bcc04d13995b8610feb5abeb3fb597bfeb7f3f00), and, actually, it looks like this branch breaks the situation. Could you confirm that I'm not going crazy? 😆

gjtorikian commented:
Eep, sorry, it's actually broken on `master` too. So neither this branch nor the PR before it worked. (Unless my test is wrong.)

obrie commented:
I just left work but will take a look later tonight.  Thanks for adding a test!

obrie commented:
@gjtorikian There were a fix issues with the test.  I've fixed that test and added another one to verify that the empty lines are actually getting removed from the html.  You'll notice that the new test also fails in master but passes in this branch.

gjtorikian commented:
Yey, thanks!
